src/toolbars/TimeSignatureToolBar.cpp

```diff
@@ -228,5 +249,35 @@ AttachedToolBarMenuItem sAttachment{
       for selecting a time range of audio */
    TimeSignatureToolBar::ID(), wxT("ShowTimeSignatureTB"), XXO("Time Signature Toolbar")
 };
+
+// Undo/redo handling of time signature changes
+// DV: where should this really go?
+
+struct TimeSignatureRestorer final : UndoStateExtension
```

Comment:
Revisiting the question of where this should go.

I like the idea of a `lib-beats` for other reasons, but even without that, it makes most sense to keep this registration in the file ProjectTimeSignature.cpp.

This would require you to change `lib-project-interface` to  `lib-project-history-interface` in lib-numeric-formats/CMakeLists.txt.

I don't think that is a bad dependency.  I consider lib-project and lib-project-history as low-level things (and have even thought that they might be merged).

Reply:
This suggestion stands